Civi/Sepa/Lock/SepaBatchLock.php

```diff
@@ -27,18 +27,28 @@
  */
 final class SepaBatchLock {
 
+  private static int $instanceCount = 0;
+
   private SepaAsyncBatchLock $asyncLock;
 
   private LockInterface $civiLock;
 
   private int $defaultTimeout;
 
   public function __construct(LockInterface $civiLock, SepaAsyncBatchLock $asyncLock, int $defaultTimeout) {
+    self::$instanceCount++;
     $this->civiLock = $civiLock;
     $this->asyncLock = $asyncLock;
     $this->defaultTimeout = $defaultTimeout;
   }
 
+  public function __destruct() {
+    self::$instanceCount--;
+    if (self::$instanceCount === 0) {
```

Comment:
Check for `<= 0` in case the static counter has been decreased somewhere else in the class?

Reply:
There's no other location where this counter is decreased and there cannot be any other.

Reply:
Well, theoretically … but okay, let's not overthink